File: Java/Tutorial/src/com/tutorial/loops/ForLoops.java
package com.tutorial.loops;

public class ForLoops {
    public static void main(String[] args) {
        System.out.println("=== For Loops ===");

        basicForLoops();
        nestedForLoops();
        printOdds();
        inForLoop();
    }

    private static void basicForLoops() {
        System.out.println("=== Basic For Loops ===");

        int[] numbers = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};

        int sum = 0;

        for (int index = 0; index < numbers.length; index++) {
            sum += numbers[index];

            System.out.println(numbers[index]);
        }

        System.out.println("Sum: " + sum);
    }

    private static void nestedForLoops() {
        System.out.println("\n=== Nested For Loops ===");

        // Outer loop to iterate through numbers 1-10
        for (int number = 1; number <= 12; number++) {
            // Inner loop to iterate through multipliers 1-10
            for (int multiplier = 1; multiplier <= 12; multiplier++) {
                // Print multiplication table entry in format: number * multiplier = result
                System.out.printf("%d * %d = %d%n", number, multiplier, number * multiplier);
            }
        }

        /*int number = 5;

        for (int multiplier = 1; multiplier < 10; multiplier++) {
            System.out.printf("%d * %d = %d%n", number, multiplier, number * multiplier);
        }*/
    }

    private static void printOdds() {
        System.out.println("\n=== Print Odds ===");

        for (int number = 1; number <= 50; number++) {
            if (number % 2 == 1) {
                System.out.println(number);
            }
        }
    }

    private static void inForLoop() {
        System.out.println("\n=== In For Loop ===");

        int[] numbers = {1, 2, 3, 4, 5};

        int sum = 0;

        for (int number : numbers) {
            System.out.println(number);

            sum += number;
        }

        System.out.println("Sum: " + sum);
    }
}

File: JavaScript/public/scripts/loops/forLoops.js
console.log("=== For Loops ===");

var myArray = [];

for (var i = 1; i < 6; i++) {
    myArray.push(i);
}

console.log(myArray);

var ourArray = [];
var myArray = [];

for (var i = 0; i < 10; i += 2) {
    ourArray.push(i);
}

for (var i = 1; i < 10; i += 2) {
    myArray.push(i);
}

console.log(ourArray);
console.log(myArray);

var ourArray = [];
var myArray = [];

for (var i = 10; i > 0; i -= 2) {
    ourArray.push(i);
}

for (var i = 9; i > 0; i -= 2) {
    myArray.push(i);
}

console.log(ourArray);
console.log(myArray);

console.log("\n=== Iterate Through Array ===");

var myArray = [2, 3, 4, 5, 6];
var total = 0;

for (var i = 0; i < myArray.length; i++) {
    total += myArray[i];
}

console.log(total);

console.log("\n=== Nested For Loop ===");

function multiplyAll(arr) {
    var product = 1;

    for (var i = 0; i < arr.length; i++) {
        for (var j = 0; j < arr[i].length; j++) {
            product *= arr[i][j];
        }
    }

    return product;
}

var product = multiplyAll([[1, 2], [3, 4], [5, 6, 7]]);

console.log(product);